### PR TITLE
material table for users

### PR DESCRIPTION
**Pull request by attwad**
This allows for pagination.
I considered moving the tag approach as in https://material-ui.com/components/selects/#multiple-select because it looks good but figured that'd be in another PR.

### CODE REVIEW

File: verification/curator-service/ui/src/components/Users.test.tsx [hunk 1]
```diff
@@ -136,14 +135,9 @@ test('updates roles on selection', async () => {
     expect(mockedAxios.put).toHaveBeenCalledWith('/api/users/abc123', {
         roles: ['admin', 'reader', 'curator'],
     });
-    expect(await findByText('Alice Smith')).toBeInTheDocument();
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/Users.test.tsx:139]: Why remove these checks?

File: verification/curator-service/ui/src/components/Users.test.tsx [hunk 2]
```diff
@@ -188,7 +178,7 @@ test('calls callback when user is changed', async () => {
         headers: {},
     };
     mockedAxios.put.mockResolvedValueOnce(axiosPutResponse);
-    expect(functionCalledCounter).toBe(0);
+    expect(mockCallback).toHaveBeenCalledTimes(0);
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/Users.test.tsx:181]: cool!

File: verification/curator-service/ui/src/components/Users.tsx [hunk 3]
```diff
@@ -13,42 +15,22 @@ interface ListResponse {
...
-    name: string;
-    email: string;
-    roles: string[];
-}
-
 interface UsersState {
     users: User[];
     availableRoles: string[];
     url: string;
+
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/Users.tsx:22]: nit: remove empty line

File: verification/curator-service/ui/src/components/Users.tsx [hunk 4]
```diff
@@ -85,113 +66,145 @@ class Users extends React.Component<Props, UsersState> {
...
+                        })
+                    }
+                    title="Users"
+                    options={{
+                        search: false,
+                        filtering: false,
+                        sorting: false,
+                        padding: 'dense',
+                        draggable: false, // No need to be able to drag and drop headers.
+                        pageSize: 10,
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/Users.tsx:142]: I had to make some changes to get choosing a pageSize to work in the other tables, could you add that here? https://github.com/open-covid-data/healthmap-gdo-temp/pull/548/files

File: verification/curator-service/ui/src/components/Users.tsx [hunk 5]
```diff
@@ -85,113 +66,145 @@ class Users extends React.Component<Props, UsersState> {
...
-        updatedUser: User,
+        userId: string,
     ): void {
         axios
-            .put(this.state.url + updatedUser._id, {
+            .put(this.state.url + userId, {
                 roles: event.target.value,
             })
             .then(() => {
                 const updatedUsers = this.state.users.slice();
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/Users.tsx:159]: Now that it's a table can we get rid of lines 159 -> 166 (RHS) and let `this.tableRef.current.onQueryChange();` do the updating?

File: verification/curator-service/ui/src/components/Users.test.tsx (hunk 1 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/Users.test.tsx:139]: contents are checked in the cypress tests, didn't like the redundancy and the fact that we have to mock again and again the axios responses here, we're just checking that the document contains whatever we put in our mocks in unit tests which feels useless to me.

File: verification/curator-service/ui/src/components/Users.tsx (hunk 3 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/Users.tsx:22]: done.

File: verification/curator-service/ui/src/components/Users.tsx (hunk 4 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/Users.tsx:142]: ah nice, thanks for pointing it out. Done.

File: verification/curator-service/ui/src/components/Users.tsx (hunk 5 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/Users.tsx:159]: you are right, much simpler now, thanks

**Pull request merged**